### PR TITLE
Address output review feedback

### PR DESCRIPTION
**Pull request by prafullbhosale**
Feedback based on aspnet/Tooling/issues/444
cc @sayedihashimi, @DamianEdwards, @Eilon  @barrytang

### CODE REVIEW

**Comment by prafullbhosale** [comment]
Tagging aspnet/Release#4

File: src/Microsoft.Extensions.CodeGenerators.Mvc/Templates/ControllerGenerator/MvcControllerWithContext.cshtml [hunk 1]
```diff
@@ -216,31 +226,52 @@ namespace @Model.ControllerNamespace
...
-        @:public async Task<IActionResult> Edit(@Model.ModelTypeName @Model.ModelVariable)
+        @:public async Task<IActionResult> Edit(@primaryKeyShortTypeName id, [Bind("@bindString")] @Model.ModelTypeName @Model.ModelVariable)
     }
     else
     {
-        @:public IActionResult Edit(@Model.ModelTypeName @Model.ModelVariable)
+        @:public IActionResult Edit(@primaryKeyShortTypeName id, [Bind("@bindString")] @Model.ModelTypeName @Model.ModelVariable)
     }
 }        {
+            if(id != @Model.ModelVariable.@primaryKeyName)
```
Comment (Eilon) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/Templates/ControllerGenerator/MvcControllerWithContext.cshtml]: Add space after `if`

File: src/Microsoft.Extensions.CodeGenerators.Mvc/Templates/ControllerGenerator/MvcControllerWithContext.cshtml [hunk 2]
```diff
@@ -216,31 +226,52 @@ namespace @Model.ControllerNamespace
...
+            {
+                return NotFound();
+            }
             if (ModelState.IsValid)
```
Comment (Eilon) [inline src/Microsoft.Extensions.CodeGenerators.Mvc/Templates/ControllerGenerator/MvcControllerWithContext.cshtml]: Insert empty line before this `if`

**Comment by Eilon** [comment]
Overall looks good.

As a side question: could we perhaps get rid of the `use async` option and just always do async controller actions? @DamianEdwards @danroth27 @sayedihashimi @barrytang any thoughts on that?

**Comment by danroth27** [comment]
Well, I wouldn't want to *always* do async controller actions. I don't think `Index`, `About` and `Contact` in our templates and item templates need to be async. But for actions that we know are accessing a database I think opting into async actions would be ok as long as the behavior is predictable and not to magical.

**Comment by Eilon** [comment]
Talked to @danroth27 in person so I will clarify: yeah I just mean that for our database-oriented controller actions we would always do async. And we could thus remove the checkbox from any UI that we have.

**Comment by prafullbhosale** [comment]
@Eilon That sounds good to me. 
I will make the changes for making the database oriented controller actions async only. 
And will remove the checkbox from the UI. 

The other controllers will remain as they are.

**Comment by prafullbhosale** [comment]
:up::date:

cc @Rick-Anderson

**Comment by prafullbhosale** [comment]
cc @phenning

**Comment by phenning** [comment]
:shipit:

**Pull request merged**